docs: Cleanup the pausing dag section (#28021)
mistercrunch/Airflow: docs/apache-airflow/concepts/dags.rst

The dependency detector is configurable, so you can implement your own logic different than the defaults in
:class:`~airflow.serialization.serialized_objects.DependencyDetector`

> Deschedule or Delete a DAG

Author: Deschedule wasn't a word in the airflow dictionary, needed to rephrase

The dependency detector is configurable, so you can implement your own logic different than the defaults in
:class:`~airflow.serialization.serialized_objects.DependencyDetector`

> Unschedule or Delete a DAG
> --------------------------------------

Reviewer: ```suggestion
Unschedule or Delete a DAG
--------------------------
```

> Unschedule or Delete a DAG

Reviewer 2: Unschedule is a strange word.

Reviewer 2: How about "deschedule" ? That sounds much better.

Author: No strong feelings either way for me! Deschedule probably matches the latin roots better? Updated accordingly

Author: Plus the alliteration is nice 😄

Reviewer 2: Not a native speaker, but for my musical ear it does sound better.

Reviewer 3: How about "Disable" or "pause"?

> There are several ways to prevent a dag from running. DAGs can be paused, deactivated, or deleted.

Reviewer: ```suggestion
A DAG can be prevented from running by either being paused, deactivated, or deleted.
```

A DAG can be paused, which prevents scheduled runs. The Scheduler will not schedule a paused DAG, 
but you can trigger it manually via the UI. The "pause" and "unpause" actions are available

> via the UI and the API.  In the UI, you can see Paused DAGs in the ``Paused`` tab. DAGs that are un-paused

Reviewer: ```suggestion
via the UI and the API.  In the UI, you can see Paused DAGs in the ``Paused`` tab. DAGs that are unpaused
```

can be found in the ``Active`` tab.

> A DAG can also be deactivated. A deactivated DAG will not run, cannot be triggered, and won't show up in the UI but its metadata 
> and history persists in the database. If a DAG is reactivated all of its history and metadata will be present.

Reviewer: ```suggestion
A DAG can also be deactivated. A deactivated DAG will not be scheduled, cannot be triggered, and won't show up in the UI, but its metadata 
and history are still persisted in the database. All of the DAG's history and metadata will still be present when it is reactivated.
```

The dependency detector is configurable, so you can implement your own logic different than the defaults in
:class:`~airflow.serialization.serialized_objects.DependencyDetector`

Controlling DAG Execution
--------------------------

> A DAG can be prevented from running by being paused, deactivated, or deleted.

Reviewer: I wonder if it’d be worthwhile to very briefly mention how each can be done here, e.g.

```rst
A DAG can be prevented from running by being:

* *Paused*, done in the web UI’s home page.
* *Deactivated*, by removing the file containing the DAG from ``DAGS_FOLDER``.
* *Deleted*, from the web UI or API.
```

This would be useful for people using the documentation as a reference, so they don’t need to read the prose below every time.

Pausing a DAG
^^^^^^^^^^^^^

> Pausing a DAG will temporarily prevent scheduled runs from occurring.

Reviewer 4: ```suggestion
Pausing a DAG will prevent scheduled runs from occurring.
```

The Scheduler will not schedule a paused DAG, but you can still trigger it manually

> via the UI. The "pause" and "unpause" actions are available in the UI and through the API.

Reviewer 4: Might be nice to show a screenshot here

In the UI, paused DAGs can be found in the ``Paused`` tab, and unpaused DAGs can be found in the 
``Active`` tab.

Deactivating a DAG
^^^^^^^^^^^^^^^^^^

> Deactivating a DAG will prevent it from being scheduled or triggered, and it will not appear in the UI.
> However, its metadata and history will still be persisted in the database.
> When the DAG is reactivated, all of its history and metadata will still be present.
> 
> To deactivate a DAG, remove it from the ``DAGS_FOLDER``.
> When the Scheduler finds a DAG in the database but missing from the ``DAGS_FOLDER``,
> it will mark it as deactivated. The metadata and history of the DAG will be kept.
> If the DAG is added back to the ``DAGS_FOLDER``, the Scheduler will mark it as active
> and its history will be visible again. Note that the "Active" tab in the Airflow UI refers
> to DAGs that are not deactivated and not paused.

Reviewer 4: ```suggestion
To remove a DAG from the UI, remove the corresponding DAG file. Airflow calls such DAGs "deactivated" DAGs. When the scheduler detects a DAG file is missing from the ``DAGS_FOLDER`` it will hide the corresponding DAG in the UI. All metadata and history of the DAG will be preserved. When the DAG file is added back to the ``DAGS_FOLDER``, the DAG and all corresponding metadata and history will be displayed again in the UI.

A "deactivated" DAG cannot be scheduled or triggered.
```

"active" and "deactivated" are internal states and the user never gets to see these. Would focus on the behaviour from a user's perspective.

Deleting a DAG
^^^^^^^^^^^^^^

> You can delete a DAG's metadata from the database using the UI or the API.

Reviewer 4: Would be good to include a screenshot here + mention which API endpoint.

If the DAG is still in the ``DAGS_FOLDER`` when you delete its metadata, it will reappear

> when the Scheduler parses the folder. Only the DAG's historical runs will be removed.

Reviewer 4: I believe _all_ metadata, except for task logs, will be removed.
